genesis/src/lockups.rs

```diff
@@ -0,0 +1,160 @@
+//! lockups generator
+use solana_sdk::{clock::Epoch, epoch_schedule::EpochSchedule, timing::years_as_slots};
+use std::time::Duration;
+
+#[derive(Debug, Default, Clone)]
+pub struct Lockups {
+    /// where in iteration progress I am
+    expiration: usize,
+    /// number of expirations post cliff
+    expirations: usize,
+    /// fractionage offered to f() last time
+    prev_fraction: f64,
+    /// portion of total available at cliff
+    cliff_fraction: f64,
+    /// time of cliff, in epochs
+    cliff_epoch: Epoch,
+    /// number of post-cliff events
+    expiration_fraction: f64,
+    /// time between each post-cliff event
+    expiration_epochs: Epoch,
+}
+
+impl Lockups {
+    pub fn by_years(
```

Comment:
How about the name `new`?

Reply:
new_by_years()?  I was following the Duration::from_XX() pattern here.  I think new() sucks.

Reply:
Then how about `from_years`?

Reply:
there are a lot of parameters, which is why I didn't use "from"

Reply:
caring less about this, but `new` was my favorite option